app/src/main/java/org/oppia/android/app/settings/profile/ProfileEditActivityPresenter.kt

```diff
@@ -9,22 +9,28 @@ import org.oppia.android.R
 import org.oppia.android.app.activity.ActivityScope
 import org.oppia.android.app.administratorcontrols.AdministratorControlsActivity
 import org.oppia.android.app.model.ProfileId
+import org.oppia.android.app.viewmodel.ViewModelProvider
 import org.oppia.android.databinding.ProfileEditActivityBinding
 import org.oppia.android.domain.profile.ProfileManagementController
 import org.oppia.android.util.data.DataProviders.Companion.toLiveData
 import org.oppia.android.util.logging.ConsoleLogger
 import javax.inject.Inject
+import kotlin.properties.Delegates
 
 /** The presenter for [ProfileEditActivity]. */
 @ActivityScope
 class ProfileEditActivityPresenter @Inject constructor(
   private val activity: AppCompatActivity,
   private val logger: ConsoleLogger,
-  private val profileManagementController: ProfileManagementController
+  private val profileManagementController: ProfileManagementController,
+  private val viewModelProvider: ViewModelProvider<ProfileEditViewModel>
 ) {
 
-  @Inject
-  lateinit var profileEditViewModel: ProfileEditViewModel
+  private val viewModel: ProfileEditViewModel by lazy {
+    getProfileEditViewModel()
+  }
+
+  private var profileId by Delegates.notNull<Int>()
```

Comment:
You can remove this as we don't need a global variable for this case.

Reply:
Ok